### PR TITLE
Refine exception message of duplicate key

### PR DESCRIPTION
**Pull request by JohnNiang**
#### What type of PR is this?

/kind improvement
/area core
/milestone 2.3.x

#### What this PR does / why we need it:

I create another exception `DuplicateNameException` to map `org.springframework.dao.DuplicateKeyException` when creating extensions.

See https://github.com/halo-dev/halo/issues/3180 for more.

#### Which issue(s) this PR fixes:

Fixes https://github.com/halo-dev/halo/issues/3180

#### Special notes for your reviewer:

1. Try to create an user which name has already exist
2. See the error

#### Does this PR introduce a user-facing change?

```release-note
优化名称重复的错误提示
```

### CODE REVIEW

File: src/main/java/run/halo/app/infra/exception/DuplicateNameException.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package run.halo.app.infra.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+public class DuplicateNameException extends ResponseStatusException {
```
Comment (guqing) [inline src/main/java/run/halo/app/infra/exception/DuplicateNameException.java:6]: 感觉这里可以直接继承 ServerWebInputException，像其子类 org.springframework.web.server.UnsatisfiedRequestParameterException 、org.springframework.web.server.UnsatisfiedRequestParameterException 和 org.springframework.web.server.MissingRequestValueException 那样都属于请求参数上的异常

Comment (JohnNiang) [reply src/main/java/run/halo/app/infra/exception/DuplicateNameException.java:6]: ServerWebInputException 会包含一些 `org.springframework.core.MethodParameter` 字段，虽然这个字段是可以为空的，但 DuplicateNameException 完全不需要这个字段。

**Review by guqing — COMMENTED** [review]
/lgtm

**Review by ruibaby — APPROVED** [review]
/approve

**Pull request merged**